source/libraries/desktop/Butler/App/SocialLogin.hs

```diff
@@ -1,19 +1,33 @@
 module Butler.App.SocialLogin (socialLoginApp) where
 
 import Butler
+import Butler.App (withEvent)
 import Butler.Display.Session
 
-appHtml :: Monad m => AppID -> UserName -> Maybe SessionProvider -> HtmlT m ()
-appHtml wid username mProvider =
-    with div_ [wid_ wid "w", class_ "flex m-auto"] do
-        with div_ [class_ "m-auto flex flex-col items-center justify-center gap-1"] do
+newtype APPState = APPState Text deriving (Generic)
```

Comment:
Maybe we should store one secret per provider:

```suggestion
newtype APPState = APPState (Map SessionProvider Text) deriving (Generic)
```

Reply:
When another provider is used then we get another session UUID and then as the ProcessMemory storage path is scoped to a session (.butler-storage/sess-8403a610-371e-430c-8861-a4d0583235ba/app-state-1.bin) I don't see the purpose of storing the SessionProvider data into APPState. The only purpose of the APPState in this demo app is to store a piece of data that demonstrate that a logged user can recover this data between logins/logouts. Or perhaps I misunderstand something.

Reply:
I'd like as a follow up PR add a "delete account button" that demonstrate a logged user can erase its account and thus its own app state.

Reply:
Ok I see, then I'm not sure how to handle different user viewing the same SocialLogin app.